src/sql/src/plan/statement.rs

```diff
@@ -708,6 +708,13 @@ impl<'a> StatementContext<'a> {
         Ok(())
     }
 
+    pub fn require_format_json(&self) -> Result<(), PlanError> {
+        if !self.unsafe_mode() && !self.catalog.system_vars().enable_format_json() {
+            sql_bail!("`FORMAT JSON` is not enabled")
```

Comment:
Once https://github.com/MaterializeInc/materialize/pull/18864 is in, which ideally would happen before this PR, this should use potentially the same structured error, or a similar thing.